gridt/src/app/core/api.service.spec.ts

```diff
@@ -1,96 +1,169 @@
+
 import { Movement } from "./models/movement.model";
+import { Identity } from "./models/identity.model";
 import { User } from './models/user.model';
-import { AuthService } from './auth.service';
 
-import { cold } from 'jasmine-marbles';
-import { HttpClient, HttpHeaders, HttpResponse, HttpHeaderResponse, HttpErrorResponse } from '@angular/common/http';
+let mock_movements: Movement[] = [
+  {
+    id: 10129312983,
+    name: "Mock Name",
+    short_description: "Mocking the short description",
+    description: "I will mock description. You are a fool, description.",
+    subscribed: false,
+    interval: "daily"
+  },
+  {
+    id: 1283912983123,
+    name: "Another mocked movement",
+    short_description: "This one only has a short description and no long one",
+    subscribed: false,
+    interval: "twice daily"
+  },
+  {
+    id: 10023123,
+    name: "Subscribed movement",
+    short_description: "This movement is supposed to come from the server.",
+    description: "We need some way to fake a response from the server, telling the client that it \
+     is subscribed to this particular movement",
+    subscribed: true,
+    interval: "daily"
+  },
+  {
+    id: 112312983,
+    name: "Flossing",
+    short_description: "Floss every day",
+    description: "We floss every day because it is good for our theeth.",
+    subscribed: true,
+    interval: "daily"
+  }
+];
 
-let mock_movements: Movement[];
 let mock_subscriptions: Movement[];
 
-describe("ApiService", () => {
-  let service: ApiService;
-  let auth: AuthService;
-  let httpMock: HttpTestingController;
-  let httpClientStub: jasmine.SpyObj<HttpClient>;
-  let authServiceStub: jasmine.SpyObj<AuthService>;
-
-  const default_headers = {
-    headers: new HttpHeaders({
-      Authorization: "JWT aksdajskd.asdjknaskdn.asdjknakdnasjd"
-    })
-  };
-
+let mock_identity: Identity[] = [
+  {
+    id: 1,
+    username: "ALittleOne",
+    bio: "Less then one foot tall.",
+    email: "a_little@one.com",
+    avatar: "arandomstring"
+  },
+  {
+    id: 2,
+    username: "JoeFlosser",
+    bio: "Clean as ever.",
+    email: "joe@flosser.com",
+    avatar: "arandomstring"
+  },
+  {
+    id: 3,
+    username: "YoMamma",
+    bio: "When I walk by, you experience a solar eclipse.",
+    email: "YoMamma@isfat.com",
+    avatar: "arandomstring"
+  }
+];
+
+const default_headers = {
+  headers: new HttpHeaders({
+    Authorization: "JWT aksdajskd.asdjknaskdn.asdjknakdnasjd"
+  })
+};
+
+let service: ApiService;
+let auth: AuthService;
+let httpMock: HttpTestingController;
+let httpClientStub: jasmine.SpyObj<HttpClient>;
+let authServiceStub: jasmine.SpyObj<AuthService>;
+
+class authServiceStub_succes {
+  isLoggedIn$ = of(true);
+  readyAuthentication$ = of(default_headers);
+}
+
+class authServiceStub_failed {
+  isLoggedIn$ = of(false);
+  readyAuthentication$ = throwError("Can't authenticate: no credentials");
+}
+
+describe("ApiService_failed_auth", () => {
```

Comment:
This is a normal string, so you can make it a nice statement like `"ApiService when auth fails"`, so that the final sentence would become something like "ApiService when auth fails should create".